ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -273,14 +285,16 @@ def advance(self, env: EnvManager) -> int:
             for brain_name, trainer in self.trainers.items():
                 if brain_name in self.trainer_metrics:
                     self.trainer_metrics[brain_name].add_delta_step(delta_time_step)
-                trainer.add_experiences(
-                    step_info.previous_all_brain_info,
-                    step_info.current_all_brain_info,
-                    step_info.brain_name_to_action_info[brain_name].outputs,
-                )
-                trainer.process_experiences(
-                    step_info.previous_all_brain_info, step_info.current_all_brain_info
-                )
+                if brain_name in step_info.brain_name_to_action_info:
+                    trainer.add_experiences(
+                        step_info.previous_all_brain_info,
+                        step_info.current_all_brain_info,
+                        step_info.brain_name_to_action_info[brain_name].outputs,
+                    )
+                    trainer.process_experiences(
+                        step_info.previous_all_brain_info,
+                        step_info.current_all_brain_info,
+                    )
```

Comment:
This is OK for now -- but we should really just change `add_experiences` and `process_experiences` to take `BrainInfo` rather than `AllBrainInfo`